src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useCallback, useEffect } from "react";
+import { HTMLAttributes } from "react";
+import styled, { css } from "styled-components";
+import ModalContent from "./ModalContent";
+import ModalPortal from "./ModalPortal";
+
+export interface ModalProps extends HTMLAttributes<HTMLDivElement> {
+  opened: boolean;
+  onClose: () => void;
+}
+
+type ModalContainerProps = Pick<ModalProps, "opened">;
+const ModalContainer = styled.div<ModalContainerProps>`
+  width: 100%;
+  height: 100%;
+
+  position: fixed;
+  top: 0;
+  left: 0;
+
+  background-color: rgba(255, 255, 255, 0.65);
+  backdrop-filter: blur(10.5px);
+
+  transition: 0.2s;
+  transition-property: opacity, visibility;
+
+  ${({ opened }) =>
+    opened
+      ? css`
+          opacity: 1;
+          visibility: visible;
+          z-index: 9999;
+        `
+      : css`
+          opacity: 0;
+          visibility: hidden;
+        `}
+`;
+
+const Modal = ({ children, opened = false, onClose, ...props }: ModalProps) => {
+  const onKeydownClose = useCallback(
+    (event: KeyboardEvent) => {
+      if (event.key === "Escape" && onClose) {
+        onClose();
+      }
+    },
+    [onClose],
+  );
+
+  useEffect(() => {
+    if (!opened) {
+      onClose();
+    }
+
+    if (opened) {
+      window.addEventListener("keyup", onKeydownClose);
+
+      return () => {
+        window.removeEventListener("keyup", onKeydownClose);
+      };
+    }
+  }, [opened]);
```

Comment:
```suggestion
const handleKeyPress = (e: React.KeyboardEvent<HTMLDivElement>) => {
    if(e.key === "Escape") {
       onClose()
    }
}
```
window event를 사용하지 않고 이런식으로도 할 수 있지 않을까요?

Reply:
모달에 키보드 이벤트로 구현하면 모달에 포커싱이 안 되어있는 경우에는 esc를 눌러도 안 닫히더라구요!
그래서 window event로 구현했는데 혹시 어느 방향이 더 좋을까요? 저도 고민했던 포인트긴 합니다!

Reply:
그런상황때문에 저렇게 하신 거군요. 이해했습니다.
`keyup` 이벤트에 외부에서 등록된 이벤트도 연결되어 있으면 같이 동작을 하겠지만
지금 프로젝트에서는 그럴 상황은 없는 것 같으니(?) 이대로 가도 괜찮을 것 같아요